tests/ui/legacy_numeric_constants.stderr

```diff
@@ -0,0 +1,171 @@
+error: importing legacy numeric constants
+  --> $DIR/legacy_numeric_constants.rs:9:5
+   |
+LL | use std::u128 as _;
+   |     ^^^^^^^^^
+   |
+   = note: `-D clippy::legacy-numeric-constants` implied by `-D warnings`
+
+error: importing legacy numeric constants
+  --> $DIR/legacy_numeric_constants.rs:12:13
+   |
+LL |     pub use std::u128;
+   |             ^^^^^^^^^
+   |
+   = help: use the associated constants on `u128` instead at their usage
+
+error: importing a legacy numeric constant
+  --> $DIR/legacy_numeric_constants.rs:44:9
+   |
+LL |     use std::u32::MAX;
+   |         ^^^^^^^^^^^^^
+   |
+   = help: use the associated constant `u32::MAX` instead at its usage
+
+error: importing a legacy numeric constant
+  --> $DIR/legacy_numeric_constants.rs:47:9
+   |
+LL |     use std::u8::MIN;
+   |         ^^^^^^^^^^^^
+   |
+   = help: use the associated constant `u8::MIN` instead at its usage
+
+error: importing legacy numeric constants
+  --> $DIR/legacy_numeric_constants.rs:77:9
+   |
+LL |     use std::u32;
+   |         ^^^^^^^^
+   |
+   = help: use the associated constants on `u32` instead at their usage
```

Comment:
Can we just recommend removing the import altogether? Maybe something like
```
help: remove this import
note: `u32::<CONST>` will then resolve to the respective associated constant
```

Reply:
We can't because somebody could do something stupid like `use crate::u32::MAX` after it, which would then break. It would be nice if we could detect that though

Reply:
Also, we should only show the note if it isn't aliased, like `use std::u32 as u32_consts`

Reply:
Maybe something like "remove this import and use the associated constants on `u32` instead"?